cmd/guacone/cmd/github.go

```diff
@@ -0,0 +1,254 @@
+			viper.GetBool("use-csub"),
+			viper.GetBool("poll"),
+			args)
+		if err != nil {
+			fmt.Printf("unable to validate flags: %v\n", err)
+			_ = cmd.Help()
+			os.Exit(1)
+		}
+
+		// GITHUB_TOKEN is the default token name
+		ghc, err := githubclient.NewGithubClient(ctx, os.Getenv("GITHUB_TOKEN"))
+		if err != nil {
+			logger.Errorf("unable to create github client: %v", err)
+		}
+
+		// Register collector
+		// TODO(lumjjb and mlieberman85): Return this to a longer duration (~10 minutes) so as
+		// to not keep hitting Github. This will require adding triggers to get new repos as
+		// they come up from the CollectSources so that there isn't a long delay from
+		// adding new data sources.
+		collectorOpts := []github.Opt{
+			github.WithCollectDataSource(opts.dataSource),
+			github.WithClient(ghc),
+			github.WithMode(opts.githubMode),
+			github.WithSbomName(opts.sbomName),
+			github.WithWorkflowName(opts.workflowFileName),
+		}
+		if opts.poll {
+			collectorOpts = append(collectorOpts, github.WithPolling(30*time.Second))
+		}
+		if opts.ownerRepoName != "" {
+			if !strings.Contains(opts.ownerRepoName, "/") {
+				logger.Errorf("owner-repo flag must be in the format <owner>/<repo>")
+			} else {
+				ownerRepoName := strings.Split(opts.ownerRepoName, "/")
+				if len(ownerRepoName) != 2 {
+					logger.Errorf("owner-repo flag must be in the format <owner>/<repo>")
+				}
+				collectorOpts = append(collectorOpts, github.WithOwner(ownerRepoName[0]))
+				collectorOpts = append(collectorOpts, github.WithRepo(ownerRepoName[1]))
+			}
+		}
+		githubCollector, err := github.NewGithubCollector(collectorOpts...)
+		if err != nil {
+			logger.Errorf("unable to create Github collector: %v", err)
+		}
+		err = collector.RegisterDocumentCollector(githubCollector, github.GithubCollector)
+		if err != nil {
+			logger.Errorf("unable to register Github collector: %v", err)
+		}
+
+		csubClient, err := csubclient.NewClient(opts.csubClientOptions)
+		if err != nil {
+			logger.Infof("collectsub client initialization failed, this ingestion will not pull in any additional data through the collectsub service: %v", err)
+			csubClient = nil
+		} else {
+			defer csubClient.Close()
+		}
+
+		var errFound bool
+
+		emit := func(d *processor.Document) error {
+			err := ingestor.Ingest(ctx, d, opts.graphqlEndpoint, csubClient)
+
+			if err != nil {
+				errFound = true
+				return fmt.Errorf("unable to ingest document: %w", err)
+			}
+			return nil
+		}
+
+		errHandler := func(err error) bool {
+			if err == nil {
+				logger.Info("collector ended gracefully")
+				return true
+			}
+			logger.Errorf("collector ended with error: %v", err)
+			return false
+		}
+
+		cancelCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		if err := collector.Collect(cancelCtx, emit, errHandler); err != nil {
+			logger.Fatal(err)
+		}
+
+		if errFound {
+			logger.Fatalf("completed ingestion with error")
+		} else {
+			logger.Infof("completed ingestion")
+		}
```

Comment:
Add signal handling similar to the deps dev collector:
https://github.com/guacsec/guac/blob/main/cmd/guacone/cmd/deps_dev.go#L102-L129